bigframes/exceptions.py

```diff
@@ -73,3 +73,7 @@ class UnknownDataTypeWarning(Warning):
 
 class ApiDeprecationWarning(FutureWarning):
     """The API has been deprecated."""
+
+
+class IncorrectApiUsageWarning(Warning):
```

Comment:
Let's be even more specific.

```suggestion
class BadIndexerKeyWarning(Warning):
```